### DISCUSSION TITLE
Release 1.2 2024-07-04

### DISCUSSION THREAD

**Discussion by lcn2** [ioccc-src/mkiocccentry#899] [Announcements]
Fixed `txzchk(1)` for BSD (in particular NetBSD) by `#include <sys/wait.h>`.
Thanks to @SirWumpus for the report!

Changed `txzchk` TXZCHK_VERSION` "1.0.3 2024-03-02"
to `"1.0.6 2024-06-25".

Updated `bug_report.sh` script with new option `-m make` to specify path to
`make(1)`. The script now also tries to find `gmake`, first and if that fails
it will look for `make`. In the case that it is not a regular executable file it
is considered a command line error along the lines of 'command not found'. This
helps simplify the handling of the error codes as without `make(1)` the script
is essentially useless anyway (or mostly useless).

Changed `bug_report.sh` BUG_REPORT_VERSION from "1.0.3 2024-06-26""
to "1.0.4 2024-06-26"`. Man page updated for these changes.

Fixed bugs that prevented NetBSD from compiling this code.  Fixed
a number of compiler warnings under NetBSD.  Thanks to @SirWumpus
for the detailed bug report and his assistance in testing.

Fixed a bug that caused `iocccsize(1)` to crash on some systems, such
as MacOS under Apple Silicon, when given a binary blob such as the
`iocccsize` binary as input.

Changed `iocccsize` IOCCCSIZE_VERSION from "28.14 2024-01-04"
to "28.15 2024-06-27".

Updated `Makefile.example` to be more like IOCCC winning entry Makefiles.

Expanded `remarks.example.md` to include more information about
IOCCC markdown guidelines as well as more consistent with the new
IOCCC submission process.

Improve the Makefile to allow for NetBSD to run `make prep`,
`make slow_prep` and `make release`.

Changed `test_ioccc/prep.sh`  PREP_VERSION from "1.0.1 2024-03-02"
to "1.0.2 2024-07-03".

Disabled parallel make for most of the repo.

Updated `.github/workflows/codeql.yml` to use up to date actions for
checkout, node setup, Initialize CodeQL, Autobuild, and Perform CodeQL Analysis.

Added `.github/workflows/test.yml` to perform make actions such as
`make slow_prep` and `make test` as part of the GitHub workflow.

Fix `.github/dependabot.yml` to set an update schedule for GitHub Actions.

Changed MKIOCCCENTRY_REPO_VERSION from "1.1 2024-05-19"
to "1.2 2024-07-04".

<hr /><em>This discussion was created from the release <a href='https://github.com/ioccc-src/mkiocccentry/releases/tag/v1.2'>Release 1.2 2024-07-04</a>.</em>

**Comment by xexyl** [comment]
I wish I had reported the parallel build problem earlier or fixed it even as this would have prevented the confusion today. I am glad I saw it in the log and it's good it's sorted now. 

I was going to update the changelog for recent updates but it seems like you might have done so? 

(I have my distance glasses on and not my reading glasses and I didn't check the file itself but if you would like me to write any updates there I can.)

I have (I think) a minor fix in something but it's not a functional fix. I can do that tomorrow. 

It's good this is done now though!
